src/codec/framed_write.rs

```diff
@@ -126,21 +121,11 @@ where
                     Some(Next::Data(ref mut frame)) => {
                         tracing::trace!(queued_data_frame = true);
                         let mut buf = (&mut self.encoder.buf).chain(frame.payload_mut());
-                        ready!(write(
-                            &mut self.inner,
-                            self.encoder.is_write_vectored,
-                            &mut buf,
-                            cx,
-                        ))?
+                        ready!(write(&mut self.inner, &mut buf, cx,))?
                     }
                     _ => {
                         tracing::trace!(queued_data_frame = false);
-                        ready!(write(
-                            &mut self.inner,
-                            self.encoder.is_write_vectored,
-                            &mut self.encoder.buf,
-                            cx,
-                        ))?
+                        ready!(write(&mut self.inner, &mut self.encoder.buf, cx,))?
```

Comment:
similarly, if we got rid of the `write` fn, we could just call `poll_write_buf` here:
```suggestion
                        ready!(tokio_util::io::poll_write_buf(Pin::new(&mut self.inner), &mut buf, cx))?
```